### PR TITLE
samples: net: coaps_server: Increase mcast addresses

### PR DESCRIPTION
**Pull request by jukkar**
The coaps server sample application did not start because
number of IPv6 multicast addresses was too low. Increased
the address count from 2 to 5.

Jira: ZEP-2359

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

### CODE REVIEW

File: samples/net/coaps_server/prj_qemu_x86.conf [hunk 1]
```diff
@@ -14,7 +14,7 @@ CONFIG_NET_PKT_TX_COUNT=14
 CONFIG_NET_BUF_RX_COUNT=30
 CONFIG_NET_BUF_TX_COUNT=30
 CONFIG_NET_IF_UNICAST_IPV6_ADDR_COUNT=3
-CONFIG_NET_IF_MCAST_IPV6_ADDR_COUNT=2
+CONFIG_NET_IF_MCAST_IPV6_ADDR_COUNT=5
```
Comment (nashif) [inline samples/net/coaps_server/prj_qemu_x86.conf]: Can we provide some safe and generous defaults for this type of low level configuration options so we do not have to deal with them in sample configurations and just take the default?

Comment (jukkar) [inline samples/net/coaps_server/prj_qemu_x86.conf]: What this test needs is 4 addresses, so I expected your question and raised the limit by 1 :)

Comment (pfalcon) [inline samples/net/coaps_server/prj_qemu_x86.conf]: Why don't we set the Kconfig default to (e.g.) 3? Then we still need to patch tests like this, but for lesser amount ;-).

Comment (jukkar) [inline samples/net/coaps_server/prj_qemu_x86.conf]: That is certainly possible, it is just difficult to know what the limit should be, but 3 could be enough for most uses. I can put this into my todo list unless anyone wants to send patch for this.

Comment (pfalcon) [inline samples/net/coaps_server/prj_qemu_x86.conf]: I'm actually more interested in discussing/settling on the general course of action - is it better to keep Z default config bare-bones, or make it more user-friendly and be able to work OOB with more apps (despite risks this approach also has)? If we agree it makes sense, then a user story can be submitted with specific criteria, and changes can be done en masse to implement to it (all this is not a priority, but can start to discuss/think now).

Comment (jukkar) [inline samples/net/coaps_server/prj_qemu_x86.conf]: IMHO we should always have working config so overestimating the requirements would be better. The application developer needs to tweak values anyways if he/she wants to optimize the resource usage. Of course finding the good values for various config options can be tricky.

**Comment by jukkar** [comment]
Just uploaded #715 that increases default number of multicast addresses to 3.

**Review by pfalcon — APPROVED** [review]
Ok, I guess we need to merge this finally ;-)

**Pull request merged**